### PR TITLE
Notifications Form and Detail customize messages

### PR DESCRIPTION
**Pull request by MilanPospisil**
Notification form/detail customize messages section.

Also ensures the user sees detail after adding/editing.

### CODE REVIEW

File: frontend/awx/administration/notifiers/NotifierForm.tsx [hunk 1]
```diff
@@ -209,11 +265,93 @@ function NotifierForm(props: { mode: 'add' | 'edit' }) {
...
+                You may apply a number of possible variables in the message. For more information,
+                refer to the
+                <a href="https://docs.ansible.com/automation-controller/latest/html/userguide/notifications.html#create-custom-notifications">
+                  Ansible Controller Documentation.
+                </a>
+              </Trans>
+            }
+            name={'customize_messages'}
+            label={t('Customize messages...')}
+          />
```
Comment (prat98) [inline frontend/awx/administration/notifiers/NotifierForm.tsx:286]: Customize message switch clears contents of form on toggle

https://github.com/ansible/ansible-ui/assets/112966709/f849b9d7-18f2-4550-a9d1-2118c949dfb6

**Review by prat98 — COMMENTED** [review]
Work looks great! Just found a few issue I found that need to be worked out

File: frontend/awx/administration/notifiers/NotifierForm.tsx [hunk 2]
```diff
@@ -209,11 +265,93 @@ function NotifierForm(props: { mode: 'add' | 'edit' }) {
...
+}
+
+function areMessagesEmpty(data: NotificationTemplate) {
+  if (data?.messages?.error?.message) return false;
+  if (data?.messages?.started?.message) return false;
+  if (data?.messages?.success?.message) return false;
+  if (data?.messages?.workflow_approval?.approved?.message) return false;
+  if (data?.messages?.workflow_approval?.denied?.message) return false;
+  if (data?.messages?.workflow_approval?.running?.message) return false;
+  if (data?.messages?.workflow_approval?.timed_out?.message) return false;
```
[the remark this answers is not in the document]
Comment (MilanPospisil) [reply frontend/awx/administration/notifiers/NotifierForm.tsx:350]: No reason, this was autogenerated from AI :-D

File: frontend/awx/administration/notifiers/NotifierForm.tsx (hunk 1 above)
Comment (MilanPospisil) [reply frontend/awx/administration/notifiers/NotifierForm.tsx:286]: Yeah, I know, I was not sure that this is the problem, but you are right, it should be implemented. The problem is, when PageFormComponents get unmounted, they also seems to delete the data. I will have to insert the data again to the context.

File: frontend/awx/administration/notifiers/NotifierForm.tsx [hunk 3]
```diff
@@ -209,11 +265,93 @@ function NotifierForm(props: { mode: 'add' | 'edit' }) {
...
+      <PageFormTextInput<NotificationTemplate>
+        name="messages.workflow_approval.running.message"
+        label={t('Workflow pending message')}
+      />
+
+      <PageFormTextInput<NotificationTemplate>
+        name="messages.workflow_approval.timed_out.message"
+        label={t('Workflow timed out message')}
+      />
+    </>
```
[the remark this answers is not in the document]
Comment (MilanPospisil) [reply frontend/awx/administration/notifiers/NotifierForm.tsx:339]: Truth to be told, Im not sure what the body is, I should check old codebase. But it seems that old UI does not use body at all. UI is sending the body as null, receiving it also as null. I god idea that body may be the second line of the editor, but if I add second line, old UI will fail without error and in network tab, request has error that it should not contains new lines.

[the remark this answers is not in the document]
Comment (prat98) [reply frontend/awx/administration/notifiers/NotifierForm.tsx:339]: So I'm seeing that the customize messages form have the body forms fields when notification type is email, PagerDuty and Webhook.
Screenshot 2024-04-22 at 11 41 00 AM

File: frontend/awx/administration/notifiers/NotifierForm.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (prat98) [reply frontend/awx/administration/notifiers/NotifierForm.tsx:350]: I think that would be better than many if statements

File: frontend/awx/administration/notifiers/NotifierForm.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (MilanPospisil) [reply frontend/awx/administration/notifiers/NotifierForm.tsx:339]: Hmm, thanks for catching that. This form is quite complicated in this number of subform types, one can easily miss something.

File: frontend/awx/administration/notifiers/NotifierForm.tsx (hunk 1 above)
Comment (prat98) [reply frontend/awx/administration/notifiers/NotifierForm.tsx:271]: If this fix will take a long time, maybe we can close out this PR right now and we can add it as an issue on Jira to be fixed during hardening?

Comment (MilanPospisil) [reply frontend/awx/administration/notifiers/NotifierForm.tsx:271]: I hope this is fixed now :-)

Comment (prat98) [reply frontend/awx/administration/notifiers/NotifierForm.tsx:271]: Sorry just realized the issue here is fixed

**Review by prat98 — APPROVED** [review]
Changes look great! 🚀

**Pull request merged**